src/Toolbox.php

```diff
@@ -2834,11 +2834,10 @@ public static function convertTagToImage($content_text, CommonDBTM $item, $doc_d
                                 $width = $img_infos[0];
                                 $height = $img_infos[1];
                             }
-
                             // Avoids creating a link within a link, when the image is already in an <a> tag
                             $add_link_tmp = $add_link;
                             if ($add_link) {
-                                $pattern = '/<a[^>]*>[^<>]*?<img[^>]+' . preg_quote($image['tag'], '/') . '[^<]+>[^<>]*?<\/a>/s';
+                                $pattern = '/<a[^>]*>(?![^<]*<a)(?:(?!<\/a>)[\s\S])*?<img[^>]*' . preg_quote($image['tag'], '/') . '[^>]*>[\s\S]*?<\/a>/s';
```

Comment:
Could you please add a comment in the code that explain all parts of this pattern ?

Reply:
Maybe this pattern would be easier to understand:
```php
// Try to detect any unclosed `<a>` tag that preced the `<img>` tag
$pattern = '/<a[^>]*>(?:(?!<\/a>).)*<img[^>]* . preg_quote($image['tag'], '/') . '/s';
```

It will detect any unclosed `<a>` tag that precede the `<img>` tag. IMHO, there is no need to try to detect the closing `</a>` placed after the `<img>` tag.